python/pyspark/ml/functions.py

```diff
@@ -106,6 +117,542 @@ def array_to_vector(col: Column) -> Column:
     return Column(sc._jvm.org.apache.spark.ml.functions.array_to_vector(_to_java_column(col)))
 
 
+def _batched(
+    data: pd.Series | pd.DataFrame | Tuple[pd.Series], batch_size: int
+) -> Iterator[pd.DataFrame]:
+    """Generator that splits a pandas dataframe/series into batches."""
+    if isinstance(data, pd.DataFrame):
+        index = 0
+        data_size = len(data)
+        while index < data_size:
+            yield data.iloc[index : index + batch_size]
+            index += batch_size
```

Comment:
The code snippet occurs in 2 places, let's make it as a function `_convert_pandas_df_to_batch_iter`

Reply:
Ended up refactoring the code a bit to simplify this.